apps/passport-client/pages/index.tsx

```diff
@@ -439,6 +440,12 @@ async function loadInitialState(): Promise<AppState> {
     serverStorageRevision: persistentSyncStatus.serverStorageRevision,
     serverStorageHash: persistentSyncStatus.serverStorageHash
   };
+
```

Comment:
I like that the validation is now a single reusable function.  For this case in particular I think there's an opportunity to validate a few more things, particularly related to what defines a valid logged in vs. logged out state.  E.g. a logged in user should have self and encryption key both set, while a logged out user should have neither.  I think it's okay for that to be a part of the broader state validation which comes later, but I think a one-off for initial load would be worth doing earlier.  I chatted with a dev who was experimenting with FrogCrypto by deleting individual localStorage fields, and this would've helped him avoid the fallout of that.

Reply:
My interpretation of this suggestion is that we should perform some extra validation here in addition to the `validateAndLogStateErrors` call (as opposed to modifying the validation function to support those extra validations), which sounds good to me. I've performed that validation inline here, but it could also be extracted to a function. Let me know if this looks good to you.

Reply:
This captures my intent, yes.  I think it should be split out into a function.  I also think this would be a good case for separately calling `validateState` so you can do your extra validation checks and add them to the same set of errors for logging.

Reply:
done